tests/integration/steps/integration_steps.go

```diff
@@ -249,6 +250,13 @@ func waitForCassDcUpgrade(t *testing.T, namespace, initialResourceVersion string
 		Namespace: namespace,
 	}
 
+	// Wait for at most 1 minute, waiting for cassdc to move to Upgrading state.
+	// This may not happen if the upgrade doesn't trigger a rolling restart.
+
+	wait.Poll(10*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
So I understand you can't use `g(t).Eventually()` because you would like to resume test execution.

But taking a step back: shouldn't the caller of this function (or the scenario being run) provide the missing information here? That is, whether we should block and wait for the state to change?

Reply:
Well, the thing is we don't know if the state will change or not. It depends on whether or not the upgrade from an older K8ssandra version to the current one will change the spec of the cassandra pods.
That's why we wait a little bit for the status change to happen and continue if it doesn't.

Reply:
OK-ish. I still find it a bit odd but don't want to hold you here. Approving.